rootex/framework/systems/post_process_system.cpp

```diff
@@ -20,3 +20,15 @@ void PostProcessSystem::update(float deltaMilliseconds)
 	RenderingDevice::GetSingleton()->unbindSRVs();
 	RenderingDevice::GetSingleton()->setOffScreenRTVDSV();
 }
+
+void PostProcessSystem::addCustomPostProcessing(const String& path)
+{
+	CameraComponent* camera = RenderSystem::GetSingleton()->getCamera();
+	m_PostProcessingDetails = camera->getPostProcessingDetails();
+
+	if (m_PostProcessingDetails.customPostProcessing.find(path) == m_PostProcessingDetails.customPostProcessing.end())
+	{
+		camera->addCustomPostProcessingDetails(path);
+		m_Processor.m_PostProcesses.emplace_back(new CustomPostProcess(path));
+	}
```

Comment:
Why is `PostProcessSystem` needing to access `CameraComponent`?
`CameraComponent` is not owning the CustomPostProcess objects, `m_PostProcess` is.

Simply check if the CustomPostProcessing effect for a the given path is already available with `m_PostProcessor` else create a new object and add it. CameraComponent only needs to know what effects are applied on a specific camera.